### PR TITLE
Validate role,subject and user claim uris under custom claim dialect

### PR DESCRIPTION
**Pull request by AnuradhaSK**
### Proposed changes in this pull request

Fix for https://github.com/wso2/product-is/issues/7911

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java [hunk 1]
```diff
@@ -357,20 +357,22 @@ private void validateClaimsConfigs(List<String> validationMsg, ClaimConfig claim
 
         ApplicationManagementService applicationMgtService = ApplicationManagementService.getInstance();
         String[] allLocalClaimUris = applicationMgtService.getAllLocalClaimUris(tenantDomain);
+        ArrayList<String> allRemoteClaimUris = new ArrayList<String>();
```
Comment (emswbandara) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: shall we rename the variable to remoteClaimUris?

also no need to define type when initializing array list. we can directly call new ArrayList<>();

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java [hunk 2]
```diff
@@ -381,6 +383,17 @@ private void validateClaimsConfigs(List<String> validationMsg, ClaimConfig claim
                     subjectClaimUri)) {
                 validationMsg.add(String.format(CLAIM_NOT_AVAILABLE, subjectClaimUri, tenantDomain));
             }
+        } else {
+            if (StringUtils.isNotBlank(roleClaimUri) && !(allRemoteClaimUris).contains(roleClaimUri)) {
+                validationMsg.add(String.format(CLAIM_NOT_AVAILABLE, roleClaimUri, tenantDomain));
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: We might need to use a different error message here.
"Local claim %s is not available in the server for tenantDomain:%s". Might not be suitable here.

Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: Is it okay to have "Service Provider claim %s is not available for Service Provider:%s." error message. 
"String serviceProviderName" will be added to **validateClaimsConfigs** method signature

Comment (mefarazath) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: Application Claim URI: xyz is not defined for Service Provider : %s 
or something along the lines would be fine :)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java (hunk 1 above)
Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: addressed in 5f538eb3b5513b3978a784dd8a0df158de48ae34

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java (hunk 2 above)
Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: addressed in 5f538eb3b5513b3978a784dd8a0df158de48ae34

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java [hunk 3]
```diff
@@ -77,6 +77,8 @@
             "for tenantDomain:%s.";
     private static final String CLAIM_NOT_AVAILABLE = "Local claim %s is not available in the server " +
             "for tenantDomain:%s.";
+    private static final String SP_CLAIM_NOT_AVAILABLE = "Application Claim URI %s is not defined " +
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]:
```suggestion
    private static final String SP_CLAIM_NOT_AVAILABLE = "Application Claim URI '%s' is not defined " +
```

Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: addressed in 121ee605b817fd29af4b4901ec148ad73160a779

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java [hunk 4]
```diff
@@ -77,6 +77,8 @@
             "for tenantDomain:%s.";
     private static final String CLAIM_NOT_AVAILABLE = "Local claim %s is not available in the server " +
             "for tenantDomain:%s.";
+    private static final String SP_CLAIM_NOT_AVAILABLE = "Application Claim URI '%s' is not defined " +
+            "for Service Provider:%s.";
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]:
```suggestion
            "for application:%s.";
```
```suggestion
            "for Service Provider:%s.";
```

Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: addressed in b29cb4f92c6de08a226dd39ed245c1df3363f5cc

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java [hunk 5]
```diff
@@ -77,6 +77,8 @@
             "for tenantDomain:%s.";
     private static final String CLAIM_NOT_AVAILABLE = "Local claim %s is not available in the server " +
             "for tenantDomain:%s.";
+    private static final String SP_CLAIM_NOT_AVAILABLE = "Application Claim URI '%s' is not defined " +
+            "for application:%s.";
```
Comment (emswbandara) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: shall we add a space after : mark

Comment (AnuradhaSK) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java]: I didn't add a space there to keep consistency with other error logs.

**Pull request merged**